drivers/display/Kconfig.st7789v

```diff
@@ -24,4 +24,7 @@ config ST7789V_RGB888
 config ST7789V_RGB565
 	bool "RGB565"
 
+config ST7789V_BGR565
```

Comment:
This looks like a case similar to that discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/71406/files#r1601871055 where the display color order needs to be different for parallel modes than serial modes. A few questions here:
- have you verified this setting works with the display sample `samples/drivers/display`? It is *critical* that the colors are in the correct corners, check the readme of that sample for more info
- have you tested this enablement with LVGL? If so, did you need to set any LVGL specific Kconfigs, such as `LV_COLOR_16_SWAP`?

Reply:
Yeah it looks like it's the same issue. I checked with `samples/drivers/display`, it's after checking that I made this change, because I noticed that the color were only right if the color mode was `BGR565`. Is that the right way to handle it ?

For LVGL, some samples don't run, I don't know why, and `samples/subsys/display/lvlg` is only black and white, so I can't check it. Is there lvgl samples with color to check ?

Reply:
@miggazElquez See `samples/modules/lvgl/demos`

Reply:
Ok, so LVGL works without swapping bytes of color at all- which does make sense, and is similar to the other case I mentioned. In that case the other thing to check would be if the colors are indeed in the correct order on the display sample. If that is the case, I'm ok with this change- just want to make sure the display works as expected if customers attempt to use it with the display API on their own.

Reply:
![IMG_20240708_142014](https://github.com/zephyrproject-rtos/zephyr/assets/50714529/383cbff3-743f-4a38-8861-f961fc3cef3b)
This is the result of the display sample, I think it's the right result.

Reply:
Yes, that looks right